src/daemon/daemonlocalserverconnection.cpp

```diff
@@ -91,6 +95,24 @@ void DaemonLocalServerConnection::parseCommand(const QByteArray& data) {
   QString type = typeValue.toString();
 
   logger.debug() << "Command received:" << type;
+  auto session = Daemon::instance()->session();
+  if (!session->isActive()) {
+    if (type == "activate" && !session->start(m_socket)) {
+      logger.error() << "Unable to start new session."
+                        "Ignoring activate command.";
+      return;
+      // Other than "activate", only "status" command is available for any user.
```

Comment:
Should 'status' be allowed for unauthorized users?

Reply:
It is only allowed when the connection is turned off. I also added this caveat for convenience due to the current state of the code. Some more discussion on https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8313#discussion_r1362162963